src/grafana_service.ts

```diff
@@ -27,8 +27,16 @@ export async function queryByMetric(
 
   let chunkParams = Object.assign({}, params);
   while(true) {
-    chunkParams.q = metric.metricQuery.getQuery(from, to, CHUNK_SIZE, data.values.length);
-    var chunk = await queryGrafana(url, apiKey, chunkParams);
+    let query = metric.metricQuery.getQuery(from, to, CHUNK_SIZE, data.values.length);
+    var chunk;
+
+    if (metric.datasource.type === 'influxdb') {
+      chunkParams.q = query;
+      chunk = await queryGrafana(url, apiKey, chunkParams, metric);
+    } else if (metric.datasource.type === 'graphite') {
+      chunk = await queryGrafana(`${url}/${query}`, apiKey, chunkParams, metric);
+    }
```

Comment:
what if it is not influxdb nor graphite ?

Reply:
fixed